File: backend/app/routers/health.py
"""
SafeVision API - Health & Stats Router
"""

import time
from fastapi import APIRouter

from app.models import HealthResponse
from app.config import settings
from app.services.detector import detector_service

router = APIRouter(tags=["Health"])

_start_time = time.time()

SUPPORTED_FORMATS = ["png", "jpg", "jpeg", "gif", "bmp", "tiff", "webp"]


@router.get(
    "/health",
    response_model=HealthResponse,
    summary="Health check",
    description="Returns API status, model state, and uptime.",
)
async def health_check():
    return HealthResponse(
        status="healthy" if detector_service.model_loaded else "degraded",
        model_loaded=detector_service.model_loaded,
        uptime_seconds=int(time.time() - _start_time),
        version="2.0.0",
        supported_formats=SUPPORTED_FORMATS,
        max_upload_size_mb=settings.max_upload_size_mb,
    )

File: middleware/src/routes/health.ts
/**
 * /api/v1/health — checks own health + compute server health.
 */

import { Router, Request, Response } from "express";
import { checkComputeHealth } from "../services/compute";

export const healthRouter = Router();

const START_TIME = Date.now();

healthRouter.get("/health", async (_req: Request, res: Response) => {
  let computeHealth: any = null;
  let computeOk = false;

  try {
    computeHealth = await checkComputeHealth();
    computeOk = computeHealth.status === "ok";
  } catch (err) {
    computeHealth = { error: "unreachable" };
  }

  const overallOk = computeOk;

  return res.status(overallOk ? 200 : 503).json({
    status: overallOk ? "ok" : "degraded",
    model_loaded: computeOk && computeHealth?.model_loaded,
    uptime_seconds: Math.floor((Date.now() - START_TIME) / 1000),
    version: "1.0.0",
    supported_formats: ["image/jpeg", "image/png", "image/webp", "image/gif"],
    max_upload_size_mb: 50,
    compute: computeHealth,
  });
});
